services/src/main/java/org/keycloak/authentication/authenticators/x509/ValidateX509CertificateUsername.java

```diff
@@ -100,36 +99,34 @@ public void authenticate(AuthenticationFlowContext context) {
         }
         catch(ModelDuplicateException e) {
             logger.modelDuplicateException(e);
```

Comment:
Not sure why there would be a ModelDuplicateException. There's no writes to the DB are there?!

Reply:
If it's valid catch, then 500 with internalServerError